### PR TITLE
circumcircle 3D

### PR DESCRIPTION
**Pull request by stla**

### CODE REVIEW

File: src/primitives/circle.jl [hunk 1]
```diff
@@ -14,6 +14,29 @@ struct Circle{T} <: Primitive{3,T}
...
+A 3D circle passing through points `p1`, `p2` and `p3`.
+"""
+function Circle(p1::Point{3}, p2::Point{3}, p3::Point{3})
+  v12 = p2 - p1
+  v13 = p3 - p1
+  m12 = coordinates(p1 + v12/2)
+  m13 = coordinates(p1 + v13/2)
+  n⃗ = normal(Plane(p1, p2, p3))
+  F = coordinates(p1) ⋅ n⃗
+  M = transpose(hcat(n⃗, v12, v13))
```
Comment (juliohm) [inline src/primitives/circle.jl]:
```suggestion
  M = transpose([n⃗ v12 v13])
```

File: src/primitives/circle.jl [hunk 2]
```diff
@@ -14,6 +14,29 @@ struct Circle{T} <: Primitive{3,T}
   radius::T
 end
 
+"""
+    Circle(p1, p2, p3)
+
+A 3D circle passing through points `p1`, `p2` and `p3`.
```
Comment (juliohm) [inline src/primitives/circle.jl:20]: Do we have a good reference for this method?

File: src/primitives/circle.jl [hunk 3]
```diff
@@ -14,6 +14,29 @@ struct Circle{T} <: Primitive{3,T}
...
+  u = [F, m12 ⋅ v12, m13 ⋅ v13]
+  O = Point(inv(M) * u)
+  r = norm(p1 - O)
+  Circle(Plane(O, n⃗), r)
+end
+
+Circle(p1::Tuple, p2::Tuple, p3::Tuple) =
+  Circle(Point(p1), Point(p2), Point(p3))
+
+
```
Comment (juliohm) [inline src/primitives/circle.jl]:
```suggestion
```

File: src/primitives/circle.jl (hunk 2 above)
Comment (stla) [reply src/primitives/circle.jl:20]: No idea. It comes from my old archives.

Comment (juliohm) [reply src/primitives/circle.jl:20]: No problem 👍🏽

**Review by juliohm — APPROVED** [review]

**Pull request merged**